main.cpp

```diff
@@ -148,29 +183,61 @@ struct Museum
         int age = 0;
         float satisfaction = 0.5f;
 
-        float payAdmissionFee(float feeAmount) { return feeAmount; }
+        Visitor()
+        {
+            std::cout << "a Museum::Visitor named '" << name << "' was created" << std::endl; 
+        }
+
+        float payAdmissionFee(float feeAmount)
+        {
+            ++timesVisited; // bit confused why this doesn't appear to update the timesVisited int
+            std::cout   << "Museum::Vistor:payAdmissionFee() - A visitor named '" << name << "' payed an admmission fee of $"
+                        << std::fixed << std::setprecision(2) << feeAmount << " and has now visited " << timesVisited << " times" << std::endl;
+            return feeAmount;
+        }
         void viewExhibit(int exhibitNum);
         void payVendor(int vendorID, float amount);
     };
 
+    Museum();
+
     float chargeVisitor(float amountToCharge, Visitor visitor) ;
     void addOrRemoveEmployees(int numberOfEmployees) { employeeCount += numberOfEmployees; }
     void lobbyPoliticians(float bribeAmount);
 };
 
+Museum::Museum()
+{
+    std::cout << "a Museum object was created" << std::endl; 
+}
+
 float Museum::chargeVisitor(float amountToCharge, Museum::Visitor visitor)
 {
+    std::cout << "Museum::chargeVisitor() a vistor named '" << visitor.name << "' was charged $" << amountToCharge << std::endl; 
     return visitor.payAdmissionFee(amountToCharge);
 }
```

Comment:
This is a common issue with new programmers or programmers who are coming from garbage-collected languages (python, C#).
When you pass this type into this function, the instance that is used inside this function is NOT the same instance that was passed in.
That means any changes you make to the instance being used INSIDE the function are NOT kept when this function ends.
the C++ concept of References and Pointers has not been covered at this point in the course. If you wanted this to work as you are most likely expecting it to work, you would need to use references or pointers.
I don't let people use coding concepts in projects until after they've been covered in the course.
You don't need to revise these function parameters.
I just want you to be aware of the potential issue.

Reply:
got it, thanks